include/osquery/events.h

```diff
@@ -448,6 +448,28 @@ class EventSubscriberPlugin : public Plugin {
    */
   Status recordEvent(EventID& eid, EventTime time);
 
+  /**
+   * @brief Get the expiration timeout for this event type
+   *
+   * The default implementation retrieves this value from FLAGS_events_expiry.
+   * This method can be overridden to allow custom event expiration timeouts in
+   * subclasses of EventSubscriberPlugin.
+   *
+   * @return The events expiration timeout for this event type
+   */
+  virtual uint64_t getEventsExpiry();
```

Comment:
Cool! I think this works. Flags work well since you can use them as switches or as configuration `{"options": {"flag_name": value}}`.